lib/src/async/future_stream.dart

```diff
@@ -39,8 +37,8 @@ class FutureStream<T> extends Stream<T> {
       // Since [controller] is synchronous, it's likely that emitting an error
       // will cause it to be cancelled before we call close.
       if (_controller != null) {
-        _controller.addError(e, stackTrace);
-        _controller.close();
+        _controller?.addError(e, stackTrace);
```

Comment:
also doesn't need the surrounding check; but then again, there are two?  Would this be the case of `_controller!.addError()..close()`?

Reply:
You could do `_controller!..addError()..close();` but dartfmt will reformat to a three-liner. Given these have null-checks though I can kill off the conditional around this though.

Reply:
Actually I'm not sure there's a one-liner that's possible regardless of dartfmt. You could imagine that addError were implemented such that it had access to enough context to null out _controller, so a second bang is necessary, but I don't think is supported in the language.